thresholdkey_generator.go

```diff
@@ -130,43 +171,54 @@ func (this *ThresholdKeyGenerator) GenerateHidingPolynomial() error {
 	return nil
 }
 
-func (this *ThresholdKeyGenerator) ComputeShare(index int) *big.Int {
+// The secred share of the i'th authority is `f(i) mod nm`, where `f` is
+// the polynomial we generated in `GenerateHidingPolynomial` function.
+func (this *ThresholdKeyGenerator) computeShare(index int) *big.Int {
 	share := big.NewInt(0)
 	for i := 0; i < this.Threshold; i++ {
 		a := this.polynomialCoefficients[i]
+		// we index authorities from 1, that's why we do index+1 here
 		b := new(big.Int).Exp(big.NewInt(int64(index+1)), big.NewInt(int64(i)), nil)
 		tmp := new(big.Int).Mul(a, b)
 		share = new(big.Int).Add(share, tmp)
 	}
 	return new(big.Int).Mod(share, this.nm)
 }
 
-func (this *ThresholdKeyGenerator) CreateShares() []*big.Int {
+func (this *ThresholdKeyGenerator) createShares() []*big.Int {
```

Comment:
Just from a Go style perspective, it would be good to change all of these to e.g `tkg` instead of `this`.